### ISSUE TITLE
Compiler warnings about unused local value ​​and local method when using the `provide` macro

### ISSUE THREAD

**Issue by szlepermarcin** [zio/zio#8986]
When using the `provide` macro with the compiler flags: `-Xlint` `-Xfatal-warnings` `-Ywarn-macros:after` `-Wunused:locals` , the compiler reports errors

minimal example using scala-cli:
```scala
//> using scala "2.13.14"
//> using dep "dev.zio::zio::2.1.5"
//> using options -Xlint -Xfatal-warnings -Ywarn-macros:after -Wunused:locals

import zio._

object ZioProvideWarnPoc extends ZIOAppDefault {

  override def run: zio.ZIO[ZioProvideWarnPoc.Environment with zio.ZIOAppArgs with zio.Scope, Any, Any] =
    Console.printLine("hello").provide(ZLayer(ZIO.unit))
}
```

result:
```
scala-cli .
Compiling project (Scala 2.13.14, JVM (17))
[error] ./ZioProvideWarnPoc.scala:13:39
[error] local val trace$macro$2 in method run is never used
[error]     Console.printLine("hello").provide(ZLayer(ZIO.unit))
[error]                                       ^
[error] ./ZioProvideWarnPoc.scala:13:39
[error] local method compose$macro$3 in method run is never used
[error]     Console.printLine("hello").provide(ZLayer(ZIO.unit))
[error]                                       ^
Error compiling project (Scala 2.13.14, JVM (17))
```

workaround - add compiler flags:
```
-Wconf:msg=local val trace.macro.* is never used:silent
-Wconf:msg=local method compose.macro.* is never used:silent
```

**Comment by hearnadam** [comment]
https://github.com/zio/zio/pull/8938

**Comment by notxcain** [comment]
@hearnadam the issue you mentioned was fixed in 2.1.5, however we still getting these errors on the latest version.

**Comment by szlepermarcin** [comment]
@notxcain this problem could have been (partially) introduced by a fix for the issue mentioned by @hearnadam
